zoo/board_games/chess/envs/chess_env.py

```diff
@@ -195,5 +194,109 @@ def seed(self, seed: int, dynamic_seed: bool = True) -> None:
     def close(self) -> None:
         pass
 
+    def bot_action(self):
+        # Get all legal moves
+        legal_moves = list(self.board.legal_moves)
+        legal_actions = chess_utils.legal_moves(self.board)
+
+        # Create a temporary board to simulate moves
+        temp_board = self.board.copy()
+
+        best_action = None
+        best_score = -np.inf
+
+        # Evaluate each legal move
+        for action, move in zip(legal_actions, legal_moves):
+            # Simulate the move
+            temp_board.push(move)
+
+            # Evaluate the board after the move
+            score = self.evaluate_board(temp_board)
+
+            # Update best_action if the current move has a higher score
+            if score > best_score:
+                best_action = action
+                best_score = score
+
+            # Undo the move to return to the original state
+            temp_board.pop()
+
+        # Return the action with the highest score
+        return best_action
+
+    def evaluate_board(self, board):
+        # Evaluate the board based on rules
+
+        # Check if the game is over
+        outcome = board.outcome()
+        if outcome is not None:
+            # If the game is over, return a definitive score
+            if outcome.winner == chess.WHITE:
+                return 10000  # White wins
+            elif outcome.winner == chess.BLACK:
+                return -10000  # Black wins
+            else:
+                return 0  # Draw
+
+        # If the game is not over, calculate a score based on various factors
+        score = 0
+
+        # 1. Assign points based on the value of pieces
+        # Piece values: Queen 9, Rook 5, Knight 3, Bishop 3, Pawn 1
+        piece_values = {
+            chess.PAWN: 1,
+            chess.KNIGHT: 3,
+            chess.BISHOP: 3,
+            chess.ROOK: 5,
+            chess.QUEEN: 9,
+            chess.KING: 0
```

Comment:
Why the value for "King" is zero? It seems strange

Reply:
In most cases, the value of the king is infinite because losing the king means the game is over. However, in this evaluation function, we do not directly assign a specific value to the king. Instead, we reflect the importance of the king through other rules (such as returning a definitive score if the king is checkmated). Assigning a value of 0 to the king can prevent the king's presence from influencing the evaluation of the position, as the number of kings does not affect the outcome when the game ends.